samples/CallWithChat/src/app/App.tsx

```diff
@@ -137,8 +147,7 @@ const generateCallWithChatArgs = async (
     locator = teamsLink;
     ensureJoinableTeamsLinkPushedToUrl(teamsLink);
   } else {
-    const callLocator: GroupCallLocator = getGroupIdFromUrl() || createGroupId();
-    ensureJoinableCallLocatorPushedToUrl(callLocator);
+    const callLocator = callLocatorGen(outboundParticipants);
```

Comment:
nit: `callLocatorGen` is doing a lot, it optionally uses a param, and it has side effects where a url might or might not be set. So it's impossible from here to know what will happen without reading the lines of code in the function.

Consider instead something like:
```tsx
let callLocator;
if (outboundParticipants) { 
  callLocator = callLocatorFromOutboundParticipants(outboundParticipants);
} else {
  callLocator = fetchGroupCallLocator();
  ensureJoinableCallLocatorPushedToUrl(callLocator);
}
```

Reply:
will update this!

Reply:
looking at the function again this was mostly done this way for conditional compilation concerns. can definitely be adjusted for when the feature is stabilized